Jenkinsfile

```diff
@@ -0,0 +1,273 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ */
+
+@Library('jenkins-joylib@v1.0.4') _
+
+pipeline {
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '30'))
+        timestamps()
+        parallelsAlwaysFailFast()
+    }
+    // Don't assign a specific agent for the entire job, in order to better
+    // share resources across jobs. Otherwise, we'd tie up an agent here for
+    // the duration of all stages for a given build, despite it not doing any
+    // actual work.
+    agent none
+
+    parameters {
+        string(
+            name: 'PLAT_CONFIGURE_ARGS',
+            defaultValue: '',
+            description:
+                'Arguments to smartos-live\'s configure script.\n' +
+                'By setting any of these, we only run the <b>"default"</b>\n' +
+                'Jenkins pipeline stage using the user-supplied value.<br/>' +
+                '<dl>\n' +
+                '<dt>-c</dt>\n' +
+                '<dd>clobber Illumos before each build [default: no]</dd>\n' +
+                '<dt>-d</dt>\n' +
+                '<dd>build Illumos in DEBUG mode only [default: no]</dd>\n' +
+                '<dt>-h</dt>\n' +
+                '<dd>this message</dd>\n' +
+                '<dt>-p gcc4</dt>\n' +
+                '<dd>primary compiler version [default: gcc7]</dd>\n' +
+                '<dt>-P password</dt>\n' +
+                '<dd>platform root password [default: randomly chosen]</dd>\n' +
+                '<dt>-r</dt>\n' +
```

Comment:
this can be removed from this list since it's always over-ridden anyway

Reply:
Ok, I can remove '-r'. Does anyone use the -P option, would you prefer if that stayed?